zerver/actions/message_delete.py

```diff
@@ -146,3 +147,44 @@ def do_delete_messages_by_sender(user: UserProfile) -> None:
     )
     if message_ids:
         move_messages_to_archive(message_ids, chunk_size=retention.STREAM_MESSAGE_BATCH_SIZE)
+
+
+def get_public_stream_messages(realm: Realm, messages: Iterable[Message]) -> list[Message]:
```

Comment:
I think this function should be in `zerver/lib/message.py` and maybe rename this something like `filter_public_stream_messages`  because we are passing messages to the function.